### PR TITLE
Tiles not loading outside of metadata bounds

### PR DESCRIPTION
**Pull request by nnhubbard**
Fixes issue #1318 where tiles outside of metadata bounds are not loading. This fix allows over zooming so that map features that are not on higher zoom levels will still show from their lower zoom level tiles.

For example, the offline mbtiles file might include world tiles in `z1` only, but when you zoom in farther those tiles are blank unless you move to the metadata bounds. With this fix, if you zoom into an area outside of the bounds you will get over zoom from `z1` which makes for a much nicer looking map without blank tiles.

### CODE REVIEW

File: platform/default/src/mbgl/storage/mbtiles_file_source.cpp [hunk 1]
```diff
@@ -209,6 +211,7 @@ class MBTilesFileSource::Impl {
                 response.noContent = false;
                 response.expires = Timestamp::max();
                 response.etag = resource.url;
+                response.error.release();
```
Comment (louwers) [inline platform/default/src/mbgl/storage/mbtiles_file_source.cpp:214]: The memory pointed to by the unique_ptr is not cleaned up after `.release()`.

**Review by louwers — CHANGES_REQUESTED** [review]
Glad you found the problem. Please see my comment.

Any chance you can add a test?

**Comment by nnhubbard** [comment]
I need to rework this pull request. Closing for now.

File: platform/default/src/mbgl/storage/mbtiles_file_source.cpp (hunk 1 above)
Comment (nnhubbard) [reply platform/default/src/mbgl/storage/mbtiles_file_source.cpp:214]: I guess I don't know how to do that. Any tips?

Comment (louwers) [reply platform/default/src/mbgl/storage/mbtiles_file_source.cpp:214]: I think you can just `reset()` instead.

Comment (nnhubbard) [reply platform/default/src/mbgl/storage/mbtiles_file_source.cpp:214]: Use `response.error.reset();` ?

**Pull request closed without merging**